java/connector-node/risingwave-sink-mock-flink/risingwave-sink-mock-flink-runtime/src/main/java/com/risingwave/mock/flink/runtime/sinkwriter/SinkWriterV2Impl.java

```diff
@@ -0,0 +1,141 @@
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.flink.api.connector.sink2.Committer;
+import org.apache.flink.api.connector.sink2.SinkWriter;
+import org.apache.flink.api.connector.sink2.StatefulSink;
+import org.apache.flink.api.connector.sink2.TwoPhaseCommittingSink;
+import org.apache.flink.table.data.RowData;
+
+/*
+ * The SinkV2-type sink will invoke this class to write RW data to the downstream.
+ * Stateful sink is currently not supported.
+ */
+public class SinkWriterV2Impl<CommT> implements com.risingwave.connector.api.sink.SinkWriter {
+
+    org.apache.flink.api.connector.sink2.Sink<RowData> sink;
+
+    SinkWriter<RowData> writer;
+
+    Committer<CommT> committer;
+
+    TableSchema tableSchema;
+
+    public SinkWriterV2Impl(
+            TableSchema tableSchema, org.apache.flink.api.connector.sink2.Sink<RowData> sink) {
+        try {
+            this.tableSchema = tableSchema;
+            this.sink = sink;
+            this.writer = sink.createWriter(new SinkWriterContextV2());
+            if (writer instanceof StatefulSink.StatefulSinkWriter) {
+                throw new UnsupportedOperationException("Don't support StatefulSink");
+            }
+            if (this.sink instanceof TwoPhaseCommittingSink) {
+                this.committer = ((TwoPhaseCommittingSink<?, CommT>) this.sink).createCommitter();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void writeRow(RowData rowData) {
+        try {
+            this.writer.write(
+                    rowData,
+                    new SinkWriter.Context() {
+                        @Override
+                        public long currentWatermark() {
+                            return 0;
+                        }
+
+                        @Override
+                        public Long timestamp() {
+                            return null;
+                        }
+                    });
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void beginEpoch(long epoch) {}
+
+    @Override
+    public boolean write(Iterable<SinkRow> rows) {
+        for (SinkRow row : rows) {
+            writeRow(new RowDataImpl(row, tableSchema.getNumColumns()));
+        }
+        return true;
+    }
+
+    @Override
+    public Optional<ConnectorServiceProto.SinkMetadata> barrier(boolean isCheckpoint) {
+        if (isCheckpoint) {
+            try {
+                this.writer.flush(false);
+                Collection<CommT> objects =
+                        ((TwoPhaseCommittingSink.PrecommittingSinkWriter<?, CommT>) this.writer)
+                                .prepareCommit();
+                if (committer != null && !objects.isEmpty()) {
```

Comment:
We should call `committer.committer` when `committer` is present no matter whether object is empty or not.

When committer is not present and object is not empty, we then throw the exception.